enclave-manager/web/src/client/KurtosisClientContext.tsx

```diff
@@ -0,0 +1,88 @@
+import { createContext, PropsWithChildren, useContext, useEffect, useState } from "react";
+import { KurtosisClient } from "./KurtosisClient";
+import { Alert, AlertDescription, AlertIcon, AlertTitle, Flex, Heading, Spinner } from "@chakra-ui/react";
+import { assertDefined, isDefined, isStringTrue, stringifyError } from "../utils";
+import { AuthenticatedKurtosisClient } from "./AuthenticatedKurtosisClient";
+import { LocalKurtosisClient } from "./LocalKurtosisClient";
+
+type KurtosisClientContextState = {
+  client: KurtosisClient | null;
+};
+
+const KurtosisClientContext = createContext<KurtosisClientContextState>({ client: null });
+
+export const KurtosisClientProvider = ({ children }: PropsWithChildren) => {
+  const [client, setClient] = useState<KurtosisClient>();
+  const [jwtToken, setJwtToken] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  useEffect(() => {
+    const receiveMessage = (event: MessageEvent) => {
+      const message = event.data.message;
+      switch (message) {
+        case "jwtToken":
+          const value = event.data.value;
+          if (isDefined(value)) {
+            setJwtToken(value);
+          }
+          break;
+      }
+    };
+    window.addEventListener("message", receiveMessage);
+    return () => window.removeEventListener("message", receiveMessage);
+  }, []);
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(window.location.search);
+    const requireAuth = isStringTrue(searchParams.get("require_authentication"));
+    const requestedApiHost = searchParams.get("api_host");
+    // eslint-disable-next-line
+    const preloadedPackage = searchParams.get("package");
```

Comment:
this will need to wire in the package into the package configuration page

Reply:
ah thanks - I wasn't really sure what this was for yet - is it a way to quicklink to a specific package for creating an enclave?

Reply:
Yes it is. You should be able to see the full code in `main` and track what it does.